File: utils/popup_handler.py

def close_popup_if_exists(page):
    popup = page.locator("#modal-intro-content")
    close_button = page.get_by_role("button", name="סגירה")

    try:
        popup.first.wait_for(state="visible", timeout=3000)

        if close_button.count() > 0:
            close_button.first.click()

    except:
        pass

File: utils/popup-handler.ts
import { Page } from "@playwright/test";

export async function closePopupIfExists(page: Page): Promise<void> {
  const popup = page.locator("#modal-intro-content");
  const closeButton = page.getByRole("button", { name: "סגירה" });

  try {
    await popup.waitFor({ state: "visible", timeout: 3000 });
    await closeButton.click();
  } catch {
    // popup לא הופיע - ממשיכים
  }
}